poetry/core/utils/helpers.py

```diff
@@ -81,6 +81,8 @@ def parse_requires(requires):  # type: (str) -> List[str]
             continue
 
         if current_marker:
+            if "@" in line:
+                line += " "
             line = "{}; {}".format(line, current_marker)
```

Comment:
```suggestion
        if current_marker:
            line = "{} ; {}".format(line, current_marker)
```

I wonder if this could introduce some edge case. I'd just add the whitespace in all cases.

Reply:
If we always add the space we break the old `parse_requires` tests (because the space was not expected).

Reply:
I am okay with the test's expected output needing updates as long it won't break any existing use cases. As far as I can tell, this change shouldn't affect anything reading metadata we generate.

Reply:
No it shouldn't, but I preferred to stick with the same representation generated here:
https://github.com/python-poetry/core/blob/e1c6978c5a0c104d373d78bcfc10a722409de583/poetry/core/_vendor/packaging/requirements.py#L123-L141

Reply:
Understandable; but this use case is a bit different as the objective here is to convert a requirement txt file into a list of requires_dist. And I suspect using `@` to check if url is not really reliable. Considering that the extra whitespace is good hygene in all cases, I'd prefer that we applied the above suggestion.

Reply:
Done

Reply:
Done